polyply/tests/test_logging.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+import logging
+import numpy as np
+import polyply
+from vermouth.log_helpers import (StyleAdapter, BipolarFormatter,
+                                  CountingHandler, TypeAdapter,)
+from polyply.tests.test_build_file_parser import test_molecule, test_system
+
+# Implement Logger
+LOGGER = TypeAdapter(logging.getLogger('polyply'))
+PRETTY_FORMATTER = logging.Formatter(fmt='{levelname:} - {type} - {message}',
+                                     style='{')
+DETAILED_FORMATTER = logging.Formatter(fmt='{levelname:} - {type} - {name} - {message}',
+                                       style='{')
+COUNTER = CountingHandler()
+
+# Control above what level message we want to count
+COUNTER.setLevel(logging.WARNING)
+
+CONSOLE_HANDLER = logging.StreamHandler()
+FORMATTER = BipolarFormatter(DETAILED_FORMATTER,
+                             PRETTY_FORMATTER,
+                             logging.DEBUG,
+                             logger=LOGGER)
+
+CONSOLE_HANDLER.setFormatter(FORMATTER)
+LOGGER.addHandler(CONSOLE_HANDLER)
+LOGGER.addHandler(COUNTER)
+
+LOGGER = StyleAdapter(LOGGER)
+
+loglevels = {0: logging.INFO, 1: logging.DEBUG, 2: 5}
+LOGGER.setLevel(loglevels[1])
```

Comment:
Isn't this a *very* strange place to define these? Don't you want to make a polyply.logging from which you import here?

Reply:
It is but it is also the only way to make pytest aware of the logging which otherwise is defined in the /bin/polyply. So you would dump this into a source file or a test fixture?

Reply:
Deduplicate it. Make a `polyply/src/logging.py` (or even in `__init__.py`) and import it in the test and in `/bin/polyply`